File: ebank/src/main/java/ma/emsi/ebank/services/MeService.java
package ma.emsi.ebank.services;

import ma.emsi.ebank.dto.account.AccountDTO;
import ma.emsi.ebank.dto.operation.OperationDTO;
import org.springframework.data.domain.Page;

import java.util.List;

public interface MeService {
    List<AccountDTO> getMyAccounts();
    Page<OperationDTO> getMyOperations(Long accountId, int page, int size);
}

File: ebank-frontend/src/services/meService.js
import { apiFetch } from "./api";

export async function getMyAccounts() {
  return apiFetch("/api/me/accounts", { method: "GET" });
}

export async function getMyOperations(accountId, page = 0, size = 10) {
  return apiFetch(
    `/api/me/accounts/${accountId}/operations?page=${page}&size=${size}`,
    {
      method: "GET",
    }
  );
}
